app/src/main/java/com/mpieterse/stride/core/services/GlobalAuthenticationListener.kt

```
@@ -0,0 +1,122 @@
+
+
+    /**
+     * Checks the validity of the current authentication state.
+     */
+    private fun checkAuthValidity() = appScope.launch {
+        try {
+            if (!firebaseAuthService.isUserSignedIn()) {
+                secureApplication()
+                return@launch
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)